backends/qualcomm/aot/python/PyQnnManagerAdaptor.h

```diff
@@ -47,20 +47,100 @@ class PyQnnManager {
     qnn_manager_ = std::make_shared<QnnManager>(
         qnn_executorch_options, qnn_executorch_context_binary_);
   }
+  // used for loading multiple graphs in qcir
+  explicit PyQnnManager(const py::bytes& buffer, const py::list& qcirs)
+      : qnn_executorch_option_ptr_(buffer) {
+    auto qnn_executorch_options = GetQnnExecuTorchOptions(
+        qnn_executorch_option_ptr_.cast<std::string_view>().data());
+
+    // merge multiple qcirs into one context with multiple graphs
+    std::vector<flatbuffers::Offset<qcir::Graph>> graphs;
+    for (size_t i = 0; i < qcirs.size(); ++i) {
+      py::buffer_info info(py::buffer(qcirs[i].cast<py::bytes>()).request());
+      flatbuffers::Verifier verifier(
+          static_cast<const uint8_t* const>(info.ptr),
+          info.size * info.itemsize);
+
+      if (!qcir::VerifyContextBuffer(verifier)) {
+        QNN_EXECUTORCH_LOG_ERROR("Fail to verify qcir format");
+        return;
+      }
+      auto context = qcir::GetContext(info.ptr);
+      for (const auto& graph : *context->graphs()) {
+        std::vector<flatbuffers::Offset<qcir::Tensor>> tensors;
+        for (const auto tensor : *graph->tensors()) {
+          // flatbuffers::Offset<Tensor> ToTensor(
+          //   QnnTensor
+          //   ToTensor(flatbuffers::Vector<::flatbuffers::Offset<qcir::Tensor>>
+          //   tensor), flatbuffers::FlatBufferBuilder* builder);
+          tensors.emplace_back(ToTensor(ToTensor(tensor), &builder_));
```

Comment:
What does `ToTensor(ToTensor(tensor), &builder_)) mean?

Reply:
I'm guessing here we'll deduplicate tensor?

Reply:
The inner `ToTensor` is used to convert serialized tensor in `qcir` to `QnnTensor` defined in QNN SDK header. The outer `ToTensor` is used to convert `QnnTensor` to flatbuffer API compatible tensor for building `qcir`.

Looks like flatbuffer has no mechanism for merging binaries, this is the detour I can come up with so far.
Will rephrase the comment for better understanding.